google-cloud-graalvm-support/src/main/java/com/google/cloud/graalvm/features/BigQueryFeature.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.graalvm.features;
+
+import static com.google.cloud.graalvm.features.NativeImageUtils.registerClassForReflection;
+import static com.google.cloud.graalvm.features.NativeImageUtils.registerClassHierarchyForReflection;
+import static com.google.cloud.graalvm.features.NativeImageUtils.registerPackageForReflection;
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import org.graalvm.nativeimage.hosted.Feature;
+
+@AutomaticFeature
+public class BigQueryFeature implements Feature {
+
+  private static final String BIG_QUERY_CLASS = "com.google.api.services.bigquery.Bigquery";
+
+  @Override
+  public void beforeAnalysis(BeforeAnalysisAccess access) {
+    Class<?> bigQueryClass = access.findClassByName(BIG_QUERY_CLASS);
+
+    if (bigQueryClass != null) {
+      registerClassHierarchyForReflection(
```

Comment:
Will compilation be faster if we only register the specific reflection-using classes?

Reply:
Yes good question -- it will be faster and the resulting image will be less memory if we registered only the ones we need.

However in this case, all of the subclasses of `com.google.api.services.bigquery.Bigquery` need to be registered it seems.